control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -247,6 +247,24 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		meta.SetStatusCondition(&hostedControlPlane.Status.Conditions, newCondition)
 	}
 
+	// Reconcile etcd restore status
+	if hostedControlPlane.Spec.Etcd.ManagementType == hyperv1.Managed && len(hostedControlPlane.Spec.Etcd.Managed.Storage.RestoreSnapshotURL) > 0 {
+		restoreCondition := meta.FindStatusCondition(hostedControlPlane.Status.Conditions, string(hyperv1.EtcdSnapshotRestored))
+		if restoreCondition == nil {
+			r.Log.Info("Reconciling etcd cluster restore status")
+			sts := manifests.EtcdStatefulSet(hostedControlPlane.Namespace)
+			if err := r.Get(ctx, client.ObjectKeyFromObject(sts), sts); err == nil {
+				newCondition := metav1.Condition{}
+				conditionPtr := r.etcdRestoredCondition(ctx, sts)
+				if conditionPtr != nil {
+					newCondition = *conditionPtr
```

Comment:
not sure why the need to copy over newCondition vs setting the retrieve cond directly?

Reply:
I just copied the same approach as `etcdStatefulSetCondition` tbh - also I wanted the option to return nil in the case where we don't have a true/false condition to set.

If you prefer I could return the condition by value from `etcdRestoredCondition` e.g with `metav1.ConditionUnknown` status in the case where we don't have a true/false status yet?